src/util/hash.rs

```diff
@@ -79,3 +79,32 @@ pub fn bitcoin_merkle_root<T, I>(mut iter: I) -> T
+        let block: Block = deserialize(&segwit_block).expect("Failed to deserialize block");
+        assert!(block.check_merkle_root()); // Sanity check.
+
+        let hashes_iter = block.txdata.iter().map(|obj| obj.txid().as_hash());
+
+        let mut hashes_vec: [sha256d::Hash; 15] = [Default::default(); 15];
```

Comment:
Maybe call this `hashes_array` and the one below `from_array`?

BTW, `default` for hash [looks weird](https://github.com/rust-bitcoin/bitcoin_hashes/issues/139) but it's not your fault and there's nothing to do about it right now.

Reply:
> Maybe call this hashes_array and the one below from_array?

Done, thanks.

> BTW, default for hash looks weird but it's not your fault and there's nothing to do about it right now.

I didn't notice at the time since I was just wanting some arbitrary value to use for initialisation before mutating but now you point it out I tend to agree.